File: src/app/document_app.py
from src.model.document import Document
from src.model.type.result import Result
from src.service.document_service import print_document_group_by_type, add_document, remove_document
from src.view.bar_view import bar
from src.view.menu_view import print_menu


def run(document_list: list[Document]) -> None:
    condition_running = True
    while condition_running:
        print_menu("Status", "Add", "Remove", "Exit")
        input_str = input("Choice: ")

        choice: int = int(input_str)
        if choice == 1:
            bar("Status")
            print_document_group_by_type(document_list)
        elif choice == 2:
            bar("Add")
            res: Result = add_document(document_list)
            print(res.message)
        elif choice == 3:
            bar("Remove")
            res: Result = remove_document(document_list)
            print(res.message)
        elif choice == 4:
            condition_running = False
        else:
            print("RETRY")

File: lib/app/document_app.rb
require_relative '../model/types/result'
require_relative '../view/bar_view'
require_relative '../view/menu_view'
require_relative '../service/document_service.rb'

def run_document_app(document_list)
  condition_running = true
  while condition_running
    print_menu('Status', 'Add', 'Remove', 'Exit')
    input_str = gets.chomp

    choice = input_str.to_i
    case choice
    when 1
      bar('Status')
      print_document_group_by_type(document_list)
    when 2
      bar('Add')
      res = add_document(document_list)
      puts res.message
    when 3
      bar('Remove')
      res = remove_document(document_list)
      puts res.message
    when 4
      condition_running = false
    else
      puts 'RETRY'
    end
  end
end
